storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10414,15 +10467,33 @@ int ha_rocksdb::inplace_populate_sk(
       }
 
       /* Create new secondary index entry */
+
+      String value_buffer;
```

Comment:
Hmm....looking at the code it seems that `Rdb_string_writer` could be modified to truncate data.  `Rdb_string_writer` is essentially just a wrapper around `std::vector<uchar>`.  So I think we can call `resize()` since we know the length of `index_flags` forsure.  

This way you can remove just the header but keep `index_flags`.  This would be another way of approaching it.